panther_utils/test/test_test_utils.cpp

```diff
@@ -58,13 +37,26 @@ TEST(TestTestUtils, CheckNanVector)
   EXPECT_THROW(TestCheckNaNVector<int>(), std::runtime_error);
 }
 
-int main(int argc, char ** argv)
+TEST(TestTestUtils, ExpectThrowWithDescription)
 {
-  rclcpp::init(argc, argv);
-  testing::InitGoogleTest(&argc, argv);
+  panther_utils::test_utils::ExpectThrowWithDescription<std::runtime_error>(
+    []() { throw std::runtime_error("Example exception"); }, "Example exception");
+
+  panther_utils::test_utils::ExpectThrowWithDescription<std::out_of_range>(
+    []() { throw std::out_of_range("Example exception"); }, "Example exception");
+
+  panther_utils::test_utils::ExpectThrowWithDescription<std::invalid_argument>(
+    []() { throw std::invalid_argument("Example exception"); }, "Example exception");
+
+  panther_utils::test_utils::ExpectThrowWithDescription<std::runtime_error>(
+    []() { throw std::runtime_error("Example exception"); }, "Example");
 
-  auto run_tests = RUN_ALL_TESTS();
+  panther_utils::test_utils::ExpectThrowWithDescription<std::runtime_error>(
+    []() { throw std::runtime_error("Example exception"); }, "exception");
+}
```

Comment:
It would be nice to add the following negative tests:
1. Throw different exception
2. Throw with the same exception but with a mismatch in error messages
3. Do not throw from a provided function